Feat/try catch emitter

tvmaasjazz opened the pull request:
#### Short description of what this resolves:
Allows for try catch decorator to be used for throwing errors, or only handling them

### PR Checklist
- [x] All `TODO`'s are done and removed
- [x] `package.json` has correct information
- [x] All dependencies are of correct type (regular vs peer vs dev)
- [x] Documented any complicated or confusing code
- [x] No linter errors
- [x] Project packages successfully (`npm pack`)
- [x] Installed local packed version in another project and tested
- [x] Updated README


#### Changes proposed in this pull request:


**JIRA Task Link:**

jpinkster commented on README.md:
```diff
@@ -67,14 +67,28 @@ redis:
 ### Decorators
 There are a few different decorators that we have made available:
-#### @TryCatch(error: Error, options?: { customResponseMessage: string } )
+#### @TryCatch(error: Error, options?: { exception: TryCatchExcepotion, handleOnly: boolean, customResponseMessage: string } )
```
```suggestion
#### @TryCatch(error: Error, options?: { exception: TryCatchException, handleOnly: boolean, customResponseMessage: string } )
```

tvmaasjazz replied:
nope, but just updated it

jpinkster commented:
@tvmaasjazz is this ready yet?

tvmaasjazz commented:
- Changed so that has optional parameters
- removed having `exception` as an option since you would just pass that in first, then set the other options in the second parameter
- added an option of `errorWrapperClass` which allows you to wrap the error in the case you are doing `handleOnly`. This comes into play for us right now if we want to silently handle an SqlException, we still want to limit the size of that error by wrapping it with the `SqlExceptionError` class

tvmaasjazz commented:
I'm going to check how easy it is getting swa and gti onto Nest 6 before rebasing this and putting these changes on master

jpinkster commented on README.md:
```diff
@@ -67,14 +67,28 @@ redis:
 ### Decorators
 There are a few different decorators that we have made available:
-#### @TryCatch(error: Error, options?: { customResponseMessage: string } )
+#### @TryCatch(error: Error, options?: { exception: TryCatchException, handleOnly: boolean, customResponseMessage: string } )
```
is this still accurate?
